tests/tag/tag.spec.tsx

```diff
@@ -0,0 +1,42 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import { Tag } from "../../src/tag";
+import { PlusCircleFillIcon } from "@lifesg/react-icons";
+
+describe("Tag", () => {
+    const label = "Some label";
+    it("should render the label correctly", () => {
+        render(
+            <Tag type="solid" data-testid="pill-component">
+                {label}
+            </Tag>
+        );
+
+        expect(screen.getByText(label)).toBeInTheDocument();
+    });
+
+    it("should render with an icon if specified", () => {
+        render(
+            <Tag
+                type="outline"
+                icon={<PlusCircleFillIcon data-testid="icon" />}
+            >
+                {label}
+            </Tag>
+        );
+
+        expect(screen.getByText(label)).toBeInTheDocument();
+        expect(screen.getByTestId("icon")).toBeInTheDocument();
+    });
+
+    it("should trigger onClick upon clicking if interactive is set as true", () => {
```

Comment:
if not interactive, do you still expect the click handler to be called? imo there's no point in blocking it (div elements have click events too), and the dev just has to remember to set the interactive flag